Add docs for v3 account deploy (#2261)
foundry-rs/starknet-foundry: docs/src/starknet/fees-and-versions.md

# `Fees` and `versions`

Historically, fees for transactions on Starknet had to be paid exclusively with ETH. However, with the rollout of v3
transactions, users now have the additional option to pay these fees using STRK.

> 💡 **Info**
> V3 transactions have additional options, that give you more control over transaction fee. You can specify the maximum gas unit price and the maximum gas for the transaction. 
This is done using the `--max-gas` and `--max-gas-unit-price` flags.

> Sncast allows you to specify in which version of transaction or fee token you want to use. This is done using

Reviewer: ```suggestion
Cast allows you to specify in which version of transaction or fee token you want to use. This is done using
```

the `--version` and `--fee-token` flags.

> 💡 **Info**
> Don't worry if you're not sure which version to use, it will be inferred automatically based on the fee token you
> provide. The same goes for the fee token, if you provide a version, the fee token will be inferred.

## sncast account deploy

> When deploying an account, you can specify the version of the transaction and the fee token to use. The following:

Reviewer: ```suggestion
When deploying an account, you can specify the version of the transaction and the fee token to use. The table below shows which token is used for which version of the transaction:
```

| Version | FeeToken |
|---------|----------|
| v1      | eth      |
| v3      | strk     |

> For strk you can do:

Reviewer: ```suggestion
When paying in STRK, you need to either set `--fee-token` to `strk`:
```

```shell
$ sncast account deploy \
    --name some-name \
    --fee-token strk \
    --max-fee 9999999999999
```

> or 

Reviewer: ```suggestion
or set `--version` to `v3`:
```

```shell
$ sncast account deploy \
    --name some-name \
    --version v3 \
    --max-fee 9999999999999
```

> For eth you can do:

Reviewer: ```suggestion
In case of paying in ETH, same rules apply. You need to set either `--fee-token` to `eth`:
```

```shell
$ sncast account deploy \
    --name some-name \
    --fee-token eth \
    --max-fee 9999999999999
```

> or 

Reviewer: ```suggestion
or set `--version` to `v1`:
```

```shell
$ sncast account deploy \
    --name some-name \
    --version v1 \
    --max-fee 9999999999999
```

> 📝 **Note**

> > When you specify max fee, it is in the smallest unit of the fee token. For ETH, it is in wei, and for STRK, it is in fri.

Reviewer: ```suggestion
> The unit used in `--max-fee` flag is the smallest unit of the given fee token. For ETH it is Wei, for STRK it is Fri.
```
